From b25e48a14c5c1da8bfba0983cffb1ac88440a421 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:24:02 -0400
Subject: [tools] perf/build: Standardize the various messages output by parallel make

Message-id: <1408462094-14194-74-git-send-email-jolsa@redhat.com>
Patchwork-id: 88046
O-Subject: [PATCH RHEL7.1 BZ1131394 073/325] tools/perf/build: Standardize the various messages output by parallel make
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 73a725f0008702600f7d987e262f963c0fa64bc6
Author: Ingo Molnar <mingo@kernel.org>
Date: Wed Oct 2 11:58:30 2013 +0200
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/Makefile b/tools/perf/Makefile
index ce7874b..3b925ad 100644
--- a/tools/perf/Makefile
+++ b/tools/perf/Makefile
@@ -13,13 +13,30 @@ endif
 
 export JOBS
 
-$(info $(shell printf '# [ perf build: Doing '\''make \033[33m-j'$(JOBS)'\033[m'\'' parallel build. ]\n'))
+define print_msg
+  @printf '    BUILD: Doing '\''make \033[33m-j'$(JOBS)'\033[m'\'' parallel build\n'
+endef
+
+define make
+  @$(MAKE) -f Makefile.perf --no-print-directory -j$(JOBS) $@
+endef
 
 #
 # Needed if no target specified:
 #
 all:
- @$(MAKE) -f Makefile.perf --no-print-directory -j$(JOBS) $@
+ $(print_msg)
+ $(make)
+
+#
+# The clean target is not really parallel, don't print the jobs info:
+#
+clean:
+ $(make)
 
+#
+# All other targets get passed through:
+#
 %:
- @$(MAKE) -f Makefile.perf --no-print-directory -j$(JOBS) $@
+ $(print_msg)
+ $(make)
diff --git a/tools/perf/config/Makefile b/tools/perf/config/Makefile
index fb6ec06..62d02cd 100644
--- a/tools/perf/config/Makefile
+++ b/tools/perf/config/Makefile
@@ -179,9 +179,9 @@ endif
 feature_print = $(eval $(feature_print_code))
 define feature_print_code
   ifeq ($(feature-$(1)), 1)
-    MSG := $(shell printf '...%30s: [ \033[32mon\033[m  ]' $(1))
+    MSG = $(shell printf '...%30s: [ \033[32mon\033[m  ]' $(1))
   else
-    MSG := $(shell printf '...%30s: [ \033[31mOFF\033[m ]' $(1))
+    MSG = $(shell printf '...%30s: [ \033[31mOFF\033[m ]' $(1))
   endif
   $(info $(MSG))
 endef
-- 
1.7.1